server/src/main/java/com/cloud/api/query/dao/NetworkOfferingJoinDaoImpl.java

```diff
@@ -54,9 +57,12 @@ protected NetworkOfferingJoinDaoImpl() {
     }
 
     @Override
-    public List<NetworkOfferingJoinVO> findByZoneId(long zoneId) {
+    public List<NetworkOfferingJoinVO> findByZoneId(long zoneId, Boolean includeAllZoneOffering) {
         SearchBuilder<NetworkOfferingJoinVO> sb = createSearchBuilder();
         sb.and("zoneId", sb.entity().getZoneId(), SearchCriteria.Op.FIND_IN_SET);
+        if (includeAllZoneOffering) {
+            sb.or("zId", sb.entity().getZoneId(), SearchCriteria.Op.NULL);
+        }
```

Comment:
the idea is zId in set or null?

Reply:
When includeAllZoneOffering is true, it should return offerings which have zoneId in DB and also offerings for which zone is not set as they will be available for all zones.